rfcs/029-resource-notices.md

```diff
@@ -0,0 +1,101 @@
+- Feature Name: Resource Notices
+- Start Date: 2021-03-22
+- RFC PR: [amundsen-io/rfcs#29](https://github.com/amundsen-io/rfcs/pull/29) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Resource Notices
+
+## Summary
+
+Resource Notices allows Amundsen administrators to show informational, warning, and alert messages related to the different resources (tables, dashboards, people) we expose in Amundsen.
+
+## Motivation
+
+This feature will help administrators show messages related to deprecation, updates (or lack of), and informational messages related to specific resources. In Lyft's case, we use them to point final users to improved versions of current surfaced tables, so final users employ more trusted data.
+
+## Guide-level Explanation (aka Product Details)
+
+An Amundsen final user can see a notice in the table/dashboard/user detail page whenever their system admin adds one related to it.
+
+A notice is a small box with an icon and a message containing HTML markup (like links and bolded text).
+
+These will come in three flavors:
+
+- Informational: Marked with a blue "i" icon on the right side
+- Warning: Marked with an orange exclamation mark icon on the right side
+- Alert: Marked with a red exclamation mark icon on the right side
+
+For example, if company X wants to deprecate the use of one table, they can opt to add a new notice in their configuration file:
+
+```
+    ... Table Configuration
+    notices: {
+        "<SCHEMANAME>.<TABLENAME>": {
+          severity: NoticeSeverity.ALERT,
+          message: `<span>This table is deprecated, please use <a href="<LINKTONEWTABLEDETAILPAGE>">this new table</a> instead.</span>`,
+        },
+    },
+```
+
+This code will show a notice with a red exclamation icon whenever a final user visits the table's Table Detail page.
+
+This feature's ultimate goal is to allow Amundsen administrators to point their users to more trusted/higher quality data without removing the old references.
+
+## Reference-level Explanation (aka Technical Details)
+
+The design we propose is based on the following:
+
+- Using our current frontend configuration files (config-utils.ts, config-default.ts, config-custom.ts, and config-types.ts)
```

Comment:
@Golodhros, The only thing I'd propose here is to use the Python/Flask config instead of JS config for this. i.e., config.py

Reasons:
- This sounds like more a metadata feature rather than a UI thing. 
- Not everyone is doing the customization of the frontend in the community. 
- For most of us, we are already doing the modification in our flask configs, so relatively easy to keep the maximum changes in one place. I'd try to minimize the places of configurations. 
- We won't need to change the frontend code in the future, as we are already serving the notices from the backend.

Reply:
I agree with you in that we should minimize the places where we have configurations, and I would love us to move into a feature flag system plus an admin UI to keep all this static configs.

I think the right place for this (for now) is the JS configuration because:
- The notices have a severity attribute that affects the looks, and we can use TypeScript for making sure they are one of the options we allow
- For now, it is not a metadata feature as we don't get the input from a third party source
- With the current state of things, both changes on the Flask config and the JS config would need a PR in the same repo, so it is basically the same without the type safety

I think that with the changes planned on the roadmap for this feature, we will end up in a place that is handy for everyone, but just for now bear with me!

Reply:
sounds good to me!